src/ui/SearchAlerts/FollowItem.ts

```diff
@@ -34,13 +42,13 @@ module Coveo {
 
       Assert.exists(this.result);
 
-      this.container = $(this.element);
-      this.text = $('<span />');
-      this.container.append(this.text);
-      this.container.on('click', $.proxy(this.toggleFollow, this));
+      this.container = $$(this.element);
+      this.text = $$('span');
+      this.container.append(this.text.el);
+      this.container.on('click', ()=>{this.toggleFollow()});
 
-      $(this.root).on(SearchAlertEvents.searchAlertDeleted, $.proxy(this.handleSubscriptionDeleted, this));
-      $(this.root).on(SearchAlertEvents.searchAlertCreated, $.proxy(this.handleSubscriptionCreated, this));
+      $$(this.root).on(SearchAlertEvents.searchAlertDeleted, (e: Event, args: ISearchAlertEventArgs)=>{this.handleSubscriptionDeleted(e, args)});

```

Comment:
When inside the scope of a Component, you should always use `this.bind.onRootElement`

`this.bind` refers to the `ComponentEvents` class, which ensure that event handler are called only if the component is not disabled. Also, the useless `Event` parameters is not passed when using that method.